homeassistant/components/sabnzbd/__init__.py

```diff
@@ -69,42 +96,79 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     return True
 
 
+@callback
+def async_get_entry_id_for_service_call(hass: HomeAssistant, call: ServiceCall) -> str:
+    """Get the entry ID related to a service call (by device ID)."""
+    api_key = call.data[ATTR_API_KEY]
+
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.title == api_key:
+            return entry.entry_id
+
+    raise ValueError(f"No api for API key: {api_key}")
+
+
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up the SabNzbd Component."""
     sab_api = await get_client(hass, entry.data)
     if not sab_api:
         raise ConfigEntryNotReady
 
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    sab_api_data = SabnzbdApiData(sab_api)
+
     hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
         KEY_API: sab_api,
+        KEY_API_DATA: sab_api_data,
```

Comment:
Please use this api data instance in the sensor platform.

Reply:
Done